docs/reference/scripting/security.asciidoc

```diff
@@ -8,6 +8,45 @@ all software has bugs and it is important to minimize the risk of failure in
 any security layer. Find below rules of thumb for how to keep Elasticsearch
 from being a vulnerability.
 
+[discrete]
+[[prefer-params]]
+=== Use parameters instead of hard-coded values
+The first time {es} sees a new script, it compiles it and stores the
+compiled version in a cache. Compilation can be a heavy process.
+
+If you need to pass variables into a script, you should pass them in as
+named `params` instead of hard-coding values into the script itself.  For
+example, if you want to be able to multiply a field value by different
+multipliers, don't hard-code the multiplier into the script:
+
+[source,js]
+----------------------
+  "source": "doc['my_field'] * 2"
+----------------------
+// NOTCONSOLE
+
+Instead, pass it in as a named parameter:
+
+[source,js]
+----------------------
+  "source": "doc['my_field'] * multiplier",
+  "params": {
+    "multiplier": 2
+  }
+----------------------
+// NOTCONSOLE
+
+The first version has to be recompiled every time the multiplier changes.  The
+second version is only compiled once.
+
+IMPORTANT: If you compile too many unique scripts within a small amount of
```

Comment:
So there's a bug in ingest pipelines that has come up a number of times related to these limits where once you hit this breaker, you have to delete the pipeline and add a new one. I wonder if should make a note of that here. @stu-elastic What do you think?

Reply:
In the ingest pipeline documentation, it's not relevant here.